packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3536,11 +3552,81 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
+  final bool _platformSupportsFadeOnScroll = switch (defaultTargetPlatform) {
+    TargetPlatform.android ||
+    TargetPlatform.iOS => true,
+    TargetPlatform.fuchsia ||
+    TargetPlatform.linux ||
+    TargetPlatform.macOS ||
+    TargetPlatform.windows => false,
+  };
+
   void _onEditableScroll() {
-    _selectionOverlay?.updateForScroll();
+    if (!_platformSupportsFadeOnScroll || _webContextMenuEnabled) {
+      _selectionOverlay?.updateForScroll();
+    }
     _scribbleCacheKey = null;
   }
 
+  void _handleContextMenuOnParentScroll(ScrollNotification notification) {
+    if (_internalScrolling) {
+      return;
+    }
+    _handleContextMenuOnScroll(notification);
+  }
+
+  Rect _calculateViewportRect() {
+    final Size screenSize = MediaQuery.sizeOf(context);
+    final ui.FlutterView view = View.of(context);
+    final double obscuredVertical = (view.padding.top + view.padding.bottom + view.viewInsets.bottom) / view.devicePixelRatio;
+    final double obscuredHorizontal = (view.padding.left + view.padding.right) / view.devicePixelRatio;
+    final Size visibleScreenSize = Size(screenSize.width - obscuredHorizontal, screenSize.height - obscuredVertical);
+    return Rect.fromLTWH(view.padding.left / view.devicePixelRatio, view.padding.top / view.devicePixelRatio, visibleScreenSize.width, visibleScreenSize.height);
+  }
+
+  void _handleContextMenuOnScroll(ScrollNotification notification) {
+    if (!_platformSupportsFadeOnScroll || _webContextMenuEnabled) {
+      return;
+    }
+    // When the scroll begins and the toolbar is visible, hide it
+    // until scrolling ends.
+    //
+    // The selection and renderEditable need to be visible within the current
+    // viewport for the toolbar to show when scrolling ends. If they are not
+    // then the toolbar is shown when they are scrolled back into view, unless
+    // invalidated by a change in TextEditingValue.
+    if (notification is ScrollStartNotification) {
+      if (_valueWhenToolbarShowScheduled != null) {
+        return;
+      }
+      final bool toolbarIsVisible = _selectionOverlay?.toolbarIsVisible ?? false;
```

Comment:
looks like `toolbarIsVisible` also accounts for the spellcheck popup.

Reply:
Are you using `_selectionOverlay?.toolbarIsVisible` to tell whether `showToolbar` was called?

Reply:
Sorry missed this, I am using it to check if the text selection toolbar is visible. `toolbarIsVisible` does account for spellcheck popup, do you recommend I add a new API that would tell us if only the text selection toolbar is visible?

Reply:
I think I can just use `!_selectionOverlay.spellCheckToolbarIsVisible` along with `toolbarIsVisible` to target the text selection toolbar.